static/src/javascripts/projects/common/modules/identity/api.js

```diff
@@ -0,0 +1,284 @@
+
+/**
+ * Gets the currently logged in user data from the identity api
+ * @param {function} callback
+ */
+export const getUserFromApi = mergeCalls(mergingCallback => {
+    const apiRoot = Id.idApiRoot ? Id.idApiRoot : '';
+
+    if (isUserLoggedIn()) {
+        ajax({
+            url: `${apiRoot}/user/me`,
+            type: 'jsonp',
+            crossOrigin: true,
+        }).then(response => {
+            if (response.status === 'ok') {
+                mergingCallback(response.user);
+            } else {
+                mergingCallback(null);
+            }
+        });
+    } else {
+        mergingCallback(null);
+    }
+});
+
+/**
+ * Clears the caches and state, primarily for testing.
+ */
+export const reset = () => {
+    getUserFromApi.reset();
+    userFromCookieCache = null;
+};
+
+/**
+ * @return {string}
+ */
+export const getCookie = () => getCookieByName(Id.cookieName);
+
+/**
+ * @return {string}
+ */
+export const getUrl = () => config.page.idUrl;
+
+/**
+ * Gets the currently logged in user data from the identity api and
+ * refreshes the users cookie at the same time.
+ */
+export const getUserFromApiWithRefreshedCookie = () => {
+    const endpoint = '/user/me';
+    const request = ajax({
+        url: Id.idApiRoot ? Id.idApiRoot + endpoint : endpoint,
+        type: 'jsonp',
+        data: {
+            refreshCookie: true,
+        },
+    });
+
+    return request;
+};
+
+/**
+ * Wrap window.location.href so it can be spied in unit tests
+ */
+export const redirectTo = (url: string): void => {
+    window.location.href = url;
+};
+
+/**
+ * Returns user object when signed in, otherwise redirects to sign in with configurable absolute returnUrl
+ */
+export const getUserOrSignIn = (paramUrl: ?string): ?Object => {
+    let returnUrl = paramUrl;
+
+    if (isUserLoggedIn()) {
+        return getUserFromCookie();
+    }
+
+    returnUrl = encodeURIComponent(returnUrl || document.location.href);
+    const url = `${getUrl() ? getUrl() : ''}/signin?returnUrl=${returnUrl}`;
+    redirectTo(url);
+};
+
+/**
+ * @return {Boolean}
+ */
+export const hasUserSignedOutInTheLast24Hours = () => {
+    const cookieData = getCookieByName(Id.signOutCookieName);
+
+    if (cookieData) {
+        return (
+            Math.round(new Date().getTime() / 1000) <
+            parseInt(cookieData, 10) + 86400
+        );
+    }
+    return false;
+};
+
+/**
+ * Returns true if a there is no signed in user and the user has not signed in the last 24 hours
+ */
+export const shouldAutoSigninInUser = function() {
+    const signedInUser = !!getCookieByName(Id.cookieName);
+    const checkFacebook = !!local.get(Id.fbCheckKey);
+    return (
+        !signedInUser &&
+        !checkFacebook &&
+        !this.hasUserSignedOutInTheLast24Hours()
+    );
+};
+
+export const setNextFbCheckTime = (nextFbCheckDue: string) => {
+    local.set(Id.fbCheckKey, {}, { expires: nextFbCheckDue });
+};
+
+export const emailSignup = (listId: string) => {
```

Comment:
This function does not appear to be used anywhere, can we delete?

Reply:
Fixed in 35f1e0794dda4010c32613397b1f644bd407fa55